lib/ruby_smb/client.rb

```diff
@@ -578,7 +578,7 @@ def recv_packet(encrypt: false)
         raw_response = dispatcher.recv_packet
       rescue RubySMB::Error::CommunicationError => e
         if encrypt
-          raise RubySMB::Error::EncryptionError, "Communication error with the "\
+          raise e, "Communication error with the "\
```

Comment:
For the context that you plan to use this in; if the session is killed - will the message be shown to the user as an encryption error?

Reply:
Good point. Should I add logic above this to check if the response is nil and raise a specific message for that scenario?
```suggestion

     raise RubySMB::Error::CommunicationError, "An error occurred reading from the Socket #{e.message}" if raw_response.nil?

      if encrypt
          raise e, "Communication error with the "\
``````

```

Reply:
I think long-term we should probably handle this better, potentially special casing this error if it's got a connection to a windows 8 target

For now, maybe we could just re-word the error message to be less clear-cut that the communication error was caused by  encryption issues

```
      if encrypt
          raise e, "Communication error with the "\
            "remote host: #{e.message}. The server supports encryption and this error may have been caused by encryption issues, but not always."
```

Or maybe we could have special case for the amount of encrypted messages sent already, and if it's more than 1 - we just don't need this message any more